android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -90,18 +90,40 @@ protected void onResume() {
         sessionRecyclerView.setAdapter(sessionsListAdapter);
         sessionRecyclerView.scrollToPosition(SessionsListAdapter.listPosition);
         sessionRecyclerView.setItemAnimator(new DefaultItemAnimator());
-
-        disposable.add(dbSingleton.getSessionsByLocationNameObservable(location)
-                .subscribe(new Consumer<ArrayList<Session>>() {
-                    @Override
-                    public void accept(@NonNull ArrayList<Session> sessions) throws Exception {
-                        mSessions.clear();
-                        mSessions.addAll(sessions);
-                        sessionsListAdapter.notifyDataSetChanged();
-
-                        handleVisibility();
-                    }
-                }));
+        if(searchText!=null){
+            //DbSingleton dbSingleton = DbSingleton.getInstance();
+
+            disposable.add(dbSingleton.getSessionsByLocationNameObservable(location)
+                    .subscribe(new Consumer<ArrayList<Session>>() {
+                        @Override
+                        public void accept(@NonNull ArrayList<Session> sessions) throws Exception {
+                            mSessions.clear();
+                            mSessions.addAll(sessions);
+
+                            final List<Session> filteredModelList = filter(mSessions,
+                                    searchText.toLowerCase(Locale.getDefault()));
+                            Timber.tag("xyz").d("%d %d", mSessions.size(), filteredModelList.size());
+
+                            sessionsListAdapter.notifyDataSetChanged();
+                            sessionsListAdapter.animateTo(filteredModelList);
+                            sessionRecyclerView.scrollToPosition(0);
+
+                            //searchText = query;
+                        }
+                    }));
+        }else {
+            disposable.add(dbSingleton.getSessionsByLocationNameObservable(location)
+                    .subscribe(new Consumer<ArrayList<Session>>() {
+                        @Override
+                        public void accept(@NonNull ArrayList<Session> sessions) throws Exception {
+                            mSessions.clear();
+                            mSessions.addAll(sessions);
+                            sessionsListAdapter.notifyDataSetChanged();
+
+                            handleVisibility();
+                        }
+                    }));
+        }
 
         handleVisibility();
```

Comment:
Do we need to call handleVisibility() again? Please tell a reason for calling it twice. Thanks.